### ISSUE TITLE
FirestorePagingAdapter - retry() failing

### ISSUE THREAD

**Issue by PatilShreyas** [firebase/FirebaseUI-Android#1608]
Hi @samtstern 
I go through [FirestorePagingAdapter](https://github.com/firebase/FirebaseUI-Android/blob/master/firestore/src/main/java/com/firebase/ui/firestore/paging/FirestorePagingAdapter.java) and there is a method in it.

```java
public void retry() {
        FirestoreDataSource source = mDataSource.getValue();
        if (source == null) {
            Log.w(TAG, "Called retry() when FirestoreDataSource is null!");
            return;
        }

        source.retry();
    }
```
I observed that `source` always remains null. Every time it shows Log...
 _"Called retry() when FirestoreDataSource is null!"_. 

I also tested calling this method from [FirestorePagingActivity](https://github.com/firebase/FirebaseUI-Android/blob/master/app/src/main/java/com/firebase/uidemo/database/firestore/FirestorePagingActivity.java) after caught in error. Still, it failed.

I implemented this in our [FirebaseRecyclerPagingAdapter](https://github.com/firebase/FirebaseUI-Android/blob/version-4.4.0-dev/database/src/main/java/com/firebase/ui/database/paging/FirebaseRecyclerPagingAdapter.java) and it is running fine!
I did it by adding `Observer` for `DataSource` and it's running fine without null value.

**Comment by samtstern** [comment]
I noticed that fix in the other adapter, thanks for that!  We should fix this in the Firestore one too, for `4.4.0`.

I'd also like to understand the underlying behavior here.  Does LiveData not hold values if nobody is observing?

**Comment by PatilShreyas** [comment]
Yes.
I observed that while working on [`FirebaseRecyclerPagingAdaper`](https://github.com/firebase/FirebaseUI-Android/blob/version-4.4.0-dev/database/src/main/java/com/firebase/ui/database/paging/FirebaseRecyclerPagingAdapter.java).

I did an experiment:
- When I added this line 
 `mDataSource.observeForever(mDataSourceObserver)`
Then after calling `getValue()` on `mDataSource` doesn't returned null value

- When I commented the code of `observeForever`
Then `mDataSource.getValue()` always returned null value.

By observing this, may be LiveData not hold values if nobody is observing

**Comment by samtstern** [comment]
Apparently if you're using `Transformations.map` or `Transformations.switchMap` then the expected behavior is for there to be no value until you add an `Observer` ... which is really tricky!

I just found this out by asking someone.

**Comment by PatilShreyas** [comment]
@samtstern Okay! 
It means we will have to add an `Observer`

**Comment by PatilShreyas** [comment]
Can I send pull request to branch `version-4.4.0-dev` by implementing this as same as `FirebaseRecyclerPagingAdapter` to fix this issue?

**Comment by samtstern** [comment]
Absolutely!

**Comment by PatilShreyas** [comment]
Should we add `refresh()` method to refresh list as `FirebaseRecyclerPagingAdapter` ?

**Comment by samtstern** [comment]
Sure I am open to that as well

**Comment by PatilShreyas** [comment]
Okay!

**Comment by PatilShreyas** [comment]
So, this issue is fixed now! 😃😃

**Comment by samtstern** [comment]
@PatilShreyas yes thank you!  I mark things as "fix-implemented" until the fix is actually released to users.

**Comment by PatilShreyas** [comment]
@samtstern Okay
Thank You!

**Comment by samtstern** [comment]
This has been fixed and released in `5.0.0`
